### DISCUSSION TITLE
Highlight out of sync blocks rather than remove translations

### DISCUSSION THREAD

**Discussion by enzedonline** [wagtail/wagtail-localize#394] [Ideas]
One for the wish list - if it's even possible:

When altering text (even if just a format change), it's really inconvenient to have to enter translations all over again on the changed text after performing the sync translations operation. Especially the case if translations are manual - I need to keep a copy of the current translation page open for each language before doing the sync, then copy over everything again. 

Repeat that for 3 languages, it takes some time and needs a lot of care. 

Can have really unpleasant results when not expecting this - it's going to be very hard to hand that over to the content editors like this.

Would be a much better route to have the out of sync blocks of text highlighted in another colour with the original translation in place still. 

Thanks :)

**Comment by kaedroho** [comment]
Thanks @enzedonline! This sounds like it could be really useful to have

In addition to this, it might be useful to be able to reuse similar translations from other pages, and I think we could implement something that supports both of these together. What do you think?

**Reply by enzedonline** [reply]
Hi @kaedroho ... sorry for the slow response, didn't see the alert.

Yes, re-useable translations would be superb as well. This is a nice-to-have for me though, not losing the existing translation when the source is updated is a real must-have for me.

Cheers!

**Comment by th3hamm0r** [comment]
Losing the old translation on sync has bitten us...

@kaedroho do you have updates on that? Is there a plan to at least keep the last translation in some way? Because as @enzedonline said, we now have to keep a backup of all translations somewhere, even when changing small typos or punctuation.

Related: https://github.com/wagtail/wagtail-localize/issues/502

**Comment by th3hamm0r** [comment]
I actually don't know in detail, how wagtail-localize works internally, but maybe displaying the old translations to a changed block is not that simple, so the following approach might be easier to implement(?):
We could try to port/implement the history view for translated pages/models too, similar to the page's history view, so the user has at least the chance to review the old state of a page/model, which can be used to copy some of the outdated strings, just to change them accordingly.

**Comment by saevarom** [comment]
I have a client with a very content rich website in two languages. We went with wagtail-localize but ultimately had to break the sync because this issue was causing too much friction and frustration.
